### PR TITLE
Imports Tests from radare2-regressions

### PR DESCRIPTION
**Pull request by thestr4ng3r**

### CODE REVIEW

**Comment by ret2libc** [comment]
+1 from me

**Comment by xarkes** [comment]
Nice!

**Comment by oddcoder** [comment]
I think this is good idea :) especially if you can keep code coverage integration (which I can't see yet) this way we can make sure that each new code commit is increasing the coverage

**Comment by thestr4ng3r** [comment]
Looks like it works well. I will remove the revert-commit before this is getting merged, so it will be in sync with master again.

**Comment by thestr4ng3r** [comment]
I hope I didn't miss anything, but this is done from my side. Please review.

File: .appveyor.yml [hunk 1]
```diff
@@ -79,10 +79,9 @@ test_script:
   - radare2 -v
   - ps: Install-Product node $env:nodejs_version
   - node --version
-  - git clone https://github.com/radareorg/radare2-regressions
-  - cd radare2-regressions/new
-  - ps: $env:pull_owner=($env:APPVEYOR_PULL_REQUEST_HEAD_REPO_NAME -split "/")[0]
-  - ps: if (-not [string]::IsNullOrWhiteSpace($env:pull_owner)) {cmd /c "(git fetch git://github.com/%pull_owner%/radare2-regressions %APPVEYOR_PULL_REQUEST_HEAD_REPO_BRANCH%:%pull_owner%_%APPVEYOR_PULL_REQUEST_HEAD_REPO_BRANCH% && git checkout %pull_owner%_%APPVEYOR_PULL_REQUEST_HEAD_REPO_BRANCH%) > nul 2>&1 || exit 0"} else {echo "Using master"}
+  - cd test
+  - git clone --depth 1 https://github.com/radareorg/radare2-testbins bins
```
Comment (XVilka) [inline .appveyor.yml]: add `-q` please

**Comment by radare** [comment]
wtf why

**Comment by thestr4ng3r** [comment]
> wtf why

Pro:
* Having the state of the tests always associated with a state of r2 source. With current r2r, it's impossible to reliably test anything except the current master unless you try to match commits by date.
* Prevents changes in tests that will accidentally break master, for example this, which made master red: https://github.com/radareorg/radare2-regressions/commit/3ab6766da4f7f1897a7dd88fd5f309e801052702
* Much easier to work with. No "create a pr in r2r with the same branch name, ..."
* Impossible to forget to merge tests for a pr
* Impossible for master to become temporarily red if multiple prs are merged in a short time period

Con:
* Adds a lot of code to this repo. But since it is all text, git handles and automatically compresses it just fine. It's also still super-small compared to the huge amount of text that is contained in shlr.

**Comment by ret2libc** [comment]
So, I tried to do a `make` from the `test` directory and it stopped here:
```
[**]                 db/anal/xtensa   OK 13122 BR 1001 XX   15 FX   23
[**]                    db/anal/z80   OK 13123 BR 1001 XX   15 FX   23
[--] fuzz binaries
ENOENT: no such file or directory, scandir './bins/fuzzed'
```

I guess the makefile is not downloading the fuzzed repository when you just run `make`.

**Comment by thestr4ng3r** [comment]
> So, I tried to do a `make` from the `test` directory and it stopped here:
> 
> ```
> [**]                 db/anal/xtensa   OK 13122 BR 1001 XX   15 FX   23
> [**]                    db/anal/z80   OK 13123 BR 1001 XX   15 FX   23
> [--] fuzz binaries
> ENOENT: no such file or directory, scandir './bins/fuzzed'
> ```
> 
> I guess the makefile is not downloading the fuzzed repository when you just run `make`.

Hmm, `bins/fuzzed` is something else than `fuzz/targets`. It will be there if all the other bins exist. Can you try with the radare2-regressions repo directly?

**Comment by thestr4ng3r** [comment]
> So, I tried to do a `make` from the `test` directory and it stopped here:
> 
> ```
> [**]                 db/anal/xtensa   OK 13122 BR 1001 XX   15 FX   23
> [**]                    db/anal/z80   OK 13123 BR 1001 XX   15 FX   23
> [--] fuzz binaries
> ENOENT: no such file or directory, scandir './bins/fuzzed'
> ```
> 
> I guess the makefile is not downloading the fuzzed repository when you just run `make`.

Yeah this issue has existed before: https://travis-ci.com/radareorg/radare2/jobs/262937727#L2816
It's because the path in r2r-js is wrong if running it from the `new/` directory.
So out of the scope of this pr.

**Comment by XVilka** [comment]
Lets do this before Microsoft put radare2 code in Arctic shelter. To not be ashamed before our grand-grand-grand kids.

File: .appveyor.yml (hunk 1 above)
Comment (thestr4ng3r) [reply .appveyor.yml]: done

File: test/.gitignore [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/bins
+/fuzz/targets
+
+/t/io/.newfile
+/t/seek/.empty.tmp
```
Comment (xarkes) [inline test/.gitignore]: If `/t/` folder is dead, are those still relevant?

Comment (thestr4ng3r) [reply test/.gitignore]: No, removed.

**Pull request merged**